homeassistant/components/deconz/__init__.py

```diff
@@ -46,44 +38,37 @@
 })
 
 
-CONFIG_INSTRUCTIONS = """
-Unlock your deCONZ gateway to register with Home Assistant.
-
-1. [Go to deCONZ system settings](http://{}:{}/edit_system.html)
-2. Press "Unlock Gateway" button
-
-[deCONZ platform documentation](https://home-assistant.io/components/deconz/)
-"""
-
-
 async def async_setup(hass, config):
-    """Set up services and configuration for deCONZ component."""
-    result = False
-    config_file = await hass.async_add_job(
-        load_json, hass.config.path(CONFIG_FILE))
-
-    async def async_deconz_discovered(service, discovery_info):
-        """Call when deCONZ gateway has been found."""
-        deconz_config = {}
-        deconz_config[CONF_HOST] = discovery_info.get(CONF_HOST)
-        deconz_config[CONF_PORT] = discovery_info.get(CONF_PORT)
-        await async_request_configuration(hass, config, deconz_config)
+    """Load configuration for deCONZ component.
 
-    if config_file:
-        result = await async_setup_deconz(hass, config, config_file)
-
-    if not result and DOMAIN in config and CONF_HOST in config[DOMAIN]:
-        deconz_config = config[DOMAIN]
-        if CONF_API_KEY in deconz_config:
-            result = await async_setup_deconz(hass, config, deconz_config)
-        else:
-            await async_request_configuration(hass, config, deconz_config)
-            return True
+    Discovery has loaded the component if DOMAIN is not present in config.
+    """
+    if DOMAIN in config:
+        deconz_config = None
+        config_file = await hass.async_add_job(
+            load_json, hass.config.path(CONFIG_FILE))
+        if config_file:
+            deconz_config = config_file
+        elif CONF_HOST in config[DOMAIN]:
+            deconz_config = config[DOMAIN]
+        if deconz_config and \
+           deconz_config[CONF_HOST] not in configured_hosts(hass):
```

Comment:
Since you accept any hostname, not just IP, this check won't work because discovered bridges will have IP. For Hue as workaround I've restricted config to just IPs

Reply:
Right now I only care for 1 bridge. But I've prepared to identify bridge regardless of hostname or IP by always storing the bridgeid.